### ISSUE TITLE
Support for OpenCV viz module?

### ISSUE THREAD

**Issue by damzam** [skvark/opencv-python#76]
I have the following installed in a virtualenv with Python 3.6.4:

opencv-python==3.4.0.12
opencv-contrib-python==3.4.0.12

on a OSX 10.13.3. I had expected the compiled viz code and corresponding Python bindings. 

https://docs.opencv.org/3.4.1/d9/d62/namespacecv_1_1viz.html

Neither of them are present. I'd never installed OpenCV on this machine. I've historically gone through the installation process manually, which isn't pleasant, and I've been able to access the viz module both via C++ and Python on Linux and Mac.

The prospect of being able to install everything without having to compile binaries is compelling, and I'd love for this to work. Is this behavior expected? If so, is there any plan to support the viz module in the future?

**Comment by skvark** [comment]
Afaik viz module requires that OpenCV is built with VTK. These packages are not built with it. 

It might be possible to enable the viz module in the future given that:

- VTK can be built or installed easily to the CI environments (Windows, macOS and CentOS 5)
- VTK license allows redistribution

**Comment by damzam** [comment]
Yep...that was definitely a requisite step when building and installing manually. And getting Qt and VTK to play nice with each other and native graphics drivers has introduced some pain points..._at least for me_.

Many thanks for the quick response!

**Comment by skvark** [comment]
I'll look into this after I've solved some issues which are currently blocking Linux builds.

**Comment by damzam** [comment]
Thanks very much. You're doing great work!

**Comment by AlejandroSilvestri** [comment]
@damzam , did you managed to use Viz on Python?

I used it with C++ on Linux, but many people use opencv with Python on Windows and they complain about not having viz.

What are the steps to get Viz running on Python?

And, because there's no documentation about vis for Python, any suggestion about how to use de methods?

Thank you

**Comment by damzam** [comment]
No. I ended up just writing the libraries I needed in C++. Sorry not be of
more help.

**Comment by asmorkalov** [comment]
Related discussion in OpenCV: https://github.com/opencv/opencv/issues/19490

**Comment by plutonium-239** [comment]
Was this ever solved?
Do I just need to build VTK and then `pip install --no-binaries opencv-contrib-python`?

**Comment by AlejandroSilvestri** [comment]
@plutonium-239 

No, it doesn't.
Python has a number of ways for 3d graphics.
I believe there's no need to adopt VTK on Python, other than easy portability from C++.
